assets/tests/components/shuttleMapPage.test.tsx

```diff
@@ -263,6 +265,40 @@ describe("Map controls", () => {
       container.querySelector("img[src^=test_satellite_url")
     ).not.toBeNull()
   })
+
+  test("on initial load, does not show user location", () => {
+    render(<ShuttleMapPage />)
+
+    expect(CurrentLocationMarker.query()).not.toBeInTheDocument()
+  })
+
+  test("after user location button is clicked, show's user location on map", async () => {
+    mockGeolocation()
+    jest
+      .mocked(navigator.geolocation.watchPosition)
```

Comment:
**suggestion:** why not mock the `useGeolocation` hook instead? I think that would be a bit simpler and also more appropriate for testing this functionality.

Reply:
I originally did this as a "just get the test working", as it was the easiest entry point. Now, I'm stuck between not wanting to mock implementation details, and wanting to do an integration test between the two. Additionally, mocking `useGeolocation` means the state sematnics don't get tested, i.e., I can't find a way to have the test pass as it is where it returns `null` before returning a `GeolocationCoordinates` object.

I'm re-reviewing the tests though so I'll see about mocking `useGeolocation` instead.

Reply:
It does cleanup nicely though 6dbb69ab1cb6d0878db40b72d5f4aa2fcf6d3e75